$DragonFly: src/gnu/usr.bin/cvs/cvs/Attic/root.c.patch,v 1.2 2005/12/07 01:16:30 corecode Exp $
Index: src/root.c
===================================================================
RCS file: /space/cvs/dragonfly/src/contrib/cvs-1.12/src/root.c,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 root.c
--- src/root.c	7 Dec 2005 00:15:13 -0000	1.1.1.1
+++ src/root.c	7 Dec 2005 01:04:02 -0000
@@ -635,10 +635,14 @@
     newroot->isremote = (newroot->method != local_method);
 
 #if defined (CLIENT_SUPPORT) || defined (SERVER_SUPPORT)
-    if (readonlyfs && newroot->isremote)
-	error (1, 0,
+    if (readonlyfs && newroot->isremote) {
+	/* downgrade readonlyfs settings via environment */
+	if (readonlyfs < 0)
+	    error (1, 0,
 "Read-only repository feature unavailable with remote roots (cvsroot = %s)",
 	       cvsroot_copy);
+	readonlyfs = 0;
+    }
 
     if ((newroot->method != local_method)
 	&& (newroot->method != fork_method)
